### PR TITLE
12 checkboxes

### LINKED ISSUE #12: Check List
It would be nice to have the ability to make items "checkable" - that is to have checkboxes in front of each item that allows a user to check off items in their list.  

* Implement some way to create new "checkable" items.
   * Create an `ItemType` enum that includes two values: `Text` and `Checkbox`
   * Add a `type` field to the `Item` class to track an item's type. By default an item's type should be `ItemType.Text`.
   * Implement some way in the UI to switch an item's type.
 * Allow the user to check/uncheck their checkbox items.
   * Persist an item's checked state in the Isar DB. 

See the Google Keep app for an example. They have the ability to make the entire list have checkboxes or not. I think it would be more powerful if our users could add normal items and checkbox items in the same list.

### PR DESCRIPTION
**Pull request by tycebrown**
Checkboxes feature added. 

Points/questions:

  - I split `SubmitValueDialog` into `EditItemDialog` and `EditListDialog`.
  - What have I named poorly? How could it be renamed? 
  - I made the `isChecked` field of `Item` to be a `bool` (with a default value of false), not a nullable `bool?`. I did this because it simplified things a bit. Would love to hear input on that decision. 
  - One last, random thing: I put the checkbox at the front of the item. Should it be at the back?

Have I prematurely optimized? Does my code need cleaning?

### CODE REVIEW

**Comment by Pertempto** [comment]
@tycebrown Is this one ready, now that I merged the previous one? Please link which one blocks which, in the comments. That way I will know without asking.

**Comment by Pertempto** [comment]
> * I split `SubmitValueDialog` into `EditItemDialog` and `EditListDialog`.  

Ok, that is fine. Please remove the `title` field from both widgets, since you already know what the value will always be for each dialog.

> * What have I named poorly? How could it be renamed?  

`EditListMetaDialog` isn't a terrible name, but I would probably prefer something like `ListSetupDialog` or `ListSettingsDialog`.


> * I made the `isChecked` field of `Item` to be a `bool` (with a default value of false), not a nullable `bool?`. I did this because it simplified things a bit. Would love to hear input on that decision.   

That sounds like a perfectly reasonable decision! It does simplify things.

> * One last, random thing: I put the checkbox at the front of the item. Should it be at the back?  

At the front makes the most sense to me! 👍🏽

File: lib/view/item_widget.dart [hunk 1]
```diff
@@ -21,14 +21,36 @@ class ItemWidget extends StatefulWidget {
...
+    TextDecoration? textDecoration;
+
+    if (widget.item.itemType == ItemType.checkbox) {
+      checkbox = Checkbox(
+        value: widget.item.isChecked,
+        onChanged: _onNewCheckedState,
+      );
+
+      textDecoration =
+          widget.item.isChecked ? TextDecoration.lineThrough : null;
```
Comment (Pertempto) [inline lib/view/item_widget.dart:37]: The strikethrough is a very nice touch!

**Comment by tycebrown** [comment]
> @tycebrown Is this one ready, now that I merged the previous one? Please link which one blocks which, in the comments. That way I will know without asking.

Sorry, yes, it's ready.

**Comment by tycebrown** [comment]
> , but I would probably prefer something like ListSetupDialog or ListSettingsDialog.

That sounds better!

**Review by Pertempto — APPROVED** [review]
Looks good!

**Pull request merged**